tests/posix/env/src/env.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <zephyr/ztest.h>
+
+#define _m_home "/home/zephyr"
+#define _m_uid  STRINGIFY(1000)
+#define _m_pwd  "/tmp"
+
+#define _m_alt_home "/this/path/is/much/longer/than" _m_home
+
+#define DEFINE_ENVIRON(_handle, _key, _val) char _handle[] = STRINGIFY(_key) "=" _val
+#define RESET_ENVIRON(_handle, _key, _val)                                                         \
+	snprintf(_handle, ARRAY_SIZE(_handle), "%s=%s", _key, _val)
+
+#if defined(CONFIG_NEWLIB_LIBC) || defined(CONFIG_PICOLIBC)
+/* newlib headers seem to be missing this */
+int getenv_r(const char *name, char *val, size_t len);
+#endif
+
+extern char **environ;
+static char **old_environ;
+
+static DEFINE_ENVIRON(home, "HOME", _m_home);
```

Comment:
Nit: These macros add a layer of complexity for no apparent benefit. 
The code is actually shorter and more readable without this macro.
```
static DEFINE_ENVIRON(home, "HOME", _m_home);
vs
static char home[] = "HOME =" _m_home;
```

Reply:
You have already demonstrated that this macro is useful :-)

Thanks for stating your preference, but I will probably keep it.